src/waveform/renderers/waveformmarkrange.cpp

```diff
@@ -75,6 +75,66 @@ WaveformMarkRange::WaveformMarkRange(
     }
 }
 
+WaveformMarkRange::WaveformMarkRange(
+        const QString& group,
+        const QColor& activeColor,
+        const QColor& disabledColor,
+        double enabledOpacity,
+        double disabledOpacity,
+        const QColor& durationTextColor,
+        const QString& startControl,
+        const QString& endControl,
+        const QString& enabledControl,
+        const QString& visibilityControl,
+        const QString& durationTextLocation)
+        : m_activeColor(activeColor),
+          m_disabledColor(disabledColor),
+          m_enabledOpacity(enabledOpacity),
+          m_disabledOpacity(disabledOpacity),
+          m_durationTextColor(durationTextColor) {
+    if (!startControl.isEmpty()) {
+        DEBUG_ASSERT(!m_markStartPointControl); // has not been created yet
+        m_markStartPointControl = std::make_unique<ControlProxy>(group, startControl);
+    }
+    if (!endControl.isEmpty()) {
+        DEBUG_ASSERT(!m_markEndPointControl); // has not been created yet
+        m_markEndPointControl = std::make_unique<ControlProxy>(group, endControl);
+    }
+
+    if (!enabledControl.isEmpty()) {
+        DEBUG_ASSERT(!m_markEnabledControl); // has not been created yet
+        m_markEnabledControl = std::make_unique<ControlProxy>(group, enabledControl);
+    }
+    if (!visibilityControl.isEmpty()) {
+        DEBUG_ASSERT(!m_markVisibleControl); // has not been created yet
+        ConfigKey key = ConfigKey::parseCommaSeparated(visibilityControl);
+        m_markVisibleControl = std::make_unique<ControlProxy>(key);
+    }
+
+    if (durationTextLocation == "before") {
+        m_durationTextLocation = DurationTextLocation::Before;
+    } else {
+        m_durationTextLocation = DurationTextLocation::After;
+    }
+
+    m_activeColor = WSkinColor::getCorrectColor(m_activeColor);
+
+    if (!m_disabledColor.isValid()) {
+        if (enabledControl.isEmpty()) {
+            m_disabledColor = QColor(Qt::transparent);
+        } else {
+            // Show warning only when there's no EnabledControl,
+            // like for intro & outro ranges.
+            QString rangeSuffix = QStringLiteral("_start_position");
+            QString rangeName = QString(startControl).remove(rangeSuffix);
+            int gray = qGray(m_activeColor.rgb());
+            m_disabledColor = QColor(gray, gray, gray);
+            qDebug() << "Didn't get DisabledColor for mark range" << rangeName
+                     << "- using desaturated Color:" << m_disabledColor;
+        }
```

Comment:
NIT: expensive string operations just for debug logging

Reply:
This operation is only ran at init time. Also, I always assumed that `qDebug` would no-op the rest of the stream of the operation if it is not enabled, isn't that right?

Reply:
Anyway, I would argue the value of this message is important as it would help indicating user that the specify color was somehow not working (e.g missing, typo, ...). Do you think I should increase the level?

Reply:
> Also, I always assumed that qDebug would no-op the rest of the stream of the operation if it is not enabled, isn't that right?

not really as far as I can tell. `rangeSuffix` and `rangeName` are still created and constructed, no matter how `qDebug` actually behaves. Also even if `qDebug` doesn't do anything, the arguments to the `operator<<` still need to be evaluated. Anyways, as I said, just a nit. I think we can achieve the same without the allocation overhead if you just implement it in terms of `QStringView`

Reply:
```suggestion
            QStringView rangeName = rangeSuffix; 
            constexpr static QStringView rangeSuffix = "_start_position";
            if (startControl.endsWith(rangeSuffix)) {
                rangeName.chop(rangeSuffix.length);
            }
            int gray = qGray(m_activeColor.rgb());
            m_disabledColor = QColor(gray, gray, gray);
            qDebug() << "Didn't get DisabledColor for mark range" << rangeName
                     << "- using desaturated Color:" << m_disabledColor;
        }
```

Reply:
Ah sorry, I just realised you weren't talking just about this `qDebug`, but the whole `else` block. I only copied this code and didn't realise that